components/server/ee/src/billing/billing-mode.ts

```diff
@@ -215,10 +210,8 @@ export class BillingModesImpl implements BillingModes {
 
         // 3. Now we're usage-based. We only have to figure out whether we have a plan yet or not.
         const result: BillingMode = { mode: "usage-based" };
-        const costCenter = await this.usageService.getCostCenter({
-            attributionId: AttributionId.render(AttributionId.create(team)),
-        });
-        if (costCenter.costCenter?.billingStrategy === CostCenter_BillingStrategy.BILLING_STRATEGY_STRIPE) {
+        const billingStrategy = await this.usageService.getCurrentBillingStategy(AttributionId.create(team));
```

Comment:
Talking to the usage service here is okay, because we should only reach this stage when `isUsageBasedBillingEnabled == true`